ratis-netty/src/main/java/org/apache/ratis/netty/server/NettyServerStreamRpc.java

```diff
@@ -115,10 +128,29 @@ private ChannelInboundHandler getServerHandler(){
       public void channelRead(ChannelHandlerContext ctx, Object msg) {
         final DataStreamRequestByteBuf request = (DataStreamRequestByteBuf)msg;
         final ByteBuf buf = request.slice();
-        final AtomicBoolean released = new AtomicBoolean();
-        streams.computeIfAbsent(request.getStreamId(), id -> getDataStreamFuture(buf, released))
-            .thenApply(stream -> writeTo(buf, stream, released.get()))
-            .thenAccept(byteWritten -> sendReply(request, byteWritten, ctx));
+        final boolean isHeader = request.getStreamOffset() == -1;
+
+        CompletableFuture<?>[] parallelWrites = new CompletableFuture<?>[clients.size() + 1];
```

Comment:
Use ArrayList, then toArray at CompletableFuture.allOf(...)

Reply:
Not doing this for two reasons
1. is this is a case that peers size is known and small, so use an fixed size array fits here.
2. is Because `CompletableFuture.allOf(...).` exposes API also using generic, java will interpret the result of that list to array as `Object` (so not `CompletableFuture<?>`), then I need to do a manual cast, which is not compile time type safe.

Reply:
Okay, reasonable